app/ota_client.py

```diff
@@ -206,25 +205,64 @@ class OtaClientUpdatePhase(Enum):
 
 class OtaClientStatistics(object):
     def __init__(self):
-        self.clear()
+        self._lock = Lock()
+        self._slot = self._init_statistics_storage()
 
-    def clear(self):
-        self.total_files = 0
-        self.files_processed = 0
+    @staticmethod
+    def _init_statistics_storage() -> dict:
+        return {
+            "total_files": 0,
+            "files_processed": 0,
+            "files_processed_copy": 0,
+            "files_processed_link": 0,
+            "files_processed_download": 0,
+            "file_size_processed_copy": 0,
+            "file_size_processed_link": 0,
+            "file_size_processed_download": 0,
+            "elapsed_time_copy": 0,  # in milliseconds
+            "elapsed_time_link": 0,  # in milliseconds
+            "elapsed_time_download": 0,  # in milliseconds
+            "errors_download": 0,
+        }
 
-        self.files_processed_copy = 0
-        self.files_processed_link = 0
-        self.files_processed_download = 0
+    def get_snapshot(self):
+        """
+        return a copy of statistics storage
+        """
+        return self._slot.copy()
 
-        self.file_size_processed_copy = 0
-        self.file_size_processed_link = 0
-        self.file_size_processed_download = 0
+    def set(self, attr: str, value):
+        """
+        set a single attr in the slot
+        """
+        with self._lock:
+            self._slot[attr] = value
 
-        self.elapsed_time_copy = 0  # in milliseconds
-        self.elapsed_time_link = 0  # in milliseconds
-        self.elapsed_time_download = 0  # in milliseconds
+    def clear(self):
+        """
+        clear the storage slot and reset to empty
+        """
+        self._slot = self._init_statistics_storage()
 
-        self.errors_download = 0
+    @contextmanager
+    def acquire_staging_storage(self):
+        """
+        acquire a staging storage for updating the slot atomically and thread-safely
+        """
+        lock_acquired = False
+        try:
+            if self._lock.acquire():
+                staging_slot: dict = self._slot.copy()
+                lock_acquired = True
+                yield staging_slot
+            else:
+                # failed to acquire lock, so we don't yield a staging_slot
+                yield None
+        finally:
+            if lock_acquired:
+                self._slot = staging_slot.copy()
+                staging_slot.clear()
+                self._lock.release()
```

Comment:
self._lock.acquire() always returns True, since timeout is not set. So if condition can be removed.
```suggestion
        try:
            self._lock.acquire()
            staging_slot: dict = self._slot.copy()
            yield staging_slot
        finally:
            self._slot = staging_slot.copy()
            staging_slot.clear()
            self._lock.release()
```

Reply:
updated in commit e3be438.